src/test/daml/Daml/Finance/Instrument/Generic/Test/ReverseConvertible.daml

```diff
@@ -0,0 +1,195 @@
+import ContingentClaims.Core.Observation (Observation(..))
+import DA.Assert ((===))
+import DA.Date (Month(..), date)
+import DA.Map qualified as M (empty, fromList)
+import DA.Set qualified as S (empty, fromList, singleton, toList)
+import DA.Time (time)
+import Daml.Finance.Data.Numeric.Observation (Observation(..))
+import Daml.Finance.Data.Time.DateClock (dateToDateClockTime)
+import Daml.Finance.Holding.Fungible qualified as Fungible (Factory(..), T)
+import Daml.Finance.Instrument.Generic.Lifecycle.Rule qualified as Lifecycle (Rule(..))
+import Daml.Finance.Instrument.Generic.Test.Util (originateGeneric)
+import Daml.Finance.Interface.Account.Util (getOwner)
+import Daml.Finance.Interface.Holding.Util (getAmount, getInstrument)
+import Daml.Finance.Interface.Lifecycle.Rule.Claim qualified as Claim (ClaimEffect(..), I)
+import Daml.Finance.Interface.Lifecycle.Rule.Lifecycle qualified as Lifecycle (Evolve(..), I)
+import Daml.Finance.Interface.Settlement.Batch qualified as Batch (Settle(..))
+import Daml.Finance.Interface.Settlement.Instruction qualified as Instruction (Allocate(..), Approve(..))
+import Daml.Finance.Interface.Settlement.Types (Allocation(..), Approval(..))
+import Daml.Finance.Interface.Types.Common.Types (Id(..), Parties)
+import Daml.Finance.Interface.Types.Date.DayCount (DayCountConventionEnum(..))
+import Daml.Finance.Lifecycle.Rule.Claim (Rule(..))
+import Daml.Finance.Settlement.Factory (Factory(..))
+import Daml.Finance.Settlement.RouteProvider.SingleCustodian (SingleCustodian(..))
+import Daml.Finance.Test.Util.Account qualified as Account (ControlledBy(..), createAccount, createFactory, credit)
+import Daml.Finance.Test.Util.Common (createParties)
+import Daml.Finance.Test.Util.Instrument qualified as Instrument (originate)
+import Daml.Finance.Test.Util.Time (createClockUpdateEvent)
+import Daml.Finance.Util.Date.DayCount (calcDcf)
+import Daml.Script
+
+-- | Parties involved in the test script.
+data TestParties = TestParties
+  with
+    bank : Party
+      -- ^ Acts as custodian in the respective holdings (both cash holdings and instrument
+      --   holdings). It also acts as depository in the reverse convertible instrument.
+    broker : Party
+      -- ^ Acts as issuer of the reverse convertible contract. It is also the party tasked with
+      --   lifecycling of the contract.
+    centralBankEU : Party
+      -- ^ The depository and the issuer of the EUR cash instrument.
+    centralBankUK : Party
+      -- ^ The depository and the issuer of the GBP cash instrument.
+    investor : Party
+      -- ^ Owner of the reverse convertible holding.
+    settlers : Parties
+      -- ^ Any of the parties can trigger the settlement of fully allocated settlement instructions.
+    publicParty : Party
+      -- ^ The public party. Every party can readAs the public party.
+
+-- Physically-settled (inverse) reverse convertible FX trade: lifecycling and settlement
```

Comment:
```suggestion
-- | Physically-settled (inverse) FX reverse convertible trade: lifecycling and settlement
```

So this is basically "inverse" because we use `EURGBP` as underlying and we win if `EURGBP` goes down (we get strike instead of the actual rate). If we were using `GBPEUR` as the underlying, then it would be a standard Reverse Convertible.

Reply:
Yes, that is the way I see it as well.